### PR TITLE
Java Solution for Multiple of 2019

### PR DESCRIPTION
**Pull request by brad182**

### CODE REVIEW

File: solutions/ac-MultipleOf2019.mdx [hunk 1]
```diff
@@ -47,5 +47,41 @@ int main()
...
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class MultipleOf2019 {
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
+		String numberString = io.next();
```
Comment (SansPapyrus683) [inline solutions/ac-MultipleOf2019.mdx:60]: just `num` would be ok

File: solutions/ac-MultipleOf2019.mdx [hunk 2]
```diff
@@ -47,5 +47,41 @@ int main()
...
+		int num = 0;
+		int stringLength = numberString.length();
+		int pow = 1;
+		int[] count = new int[2019];
+		count[0] = 1;
+
+		for (int i = stringLength - 1; i >= 0; i--) {
+			//find the remainder of the current number mod 2019
+			num = (num + pow * (numberString.charAt(i) - '0')) % 2019;
```
Comment (SansPapyrus683) [inline solutions/ac-MultipleOf2019.mdx:69]: maybe Character.getNumericValue would be more explicit?

File: solutions/ac-MultipleOf2019.mdx [hunk 3]
```diff
@@ -47,5 +47,41 @@ int main()
...
+			//increment the count of this remainder
+			count[num]++;
+			pow = pow * 10 % 2019;
+		}
+		long answer = 0;
```
Comment (SansPapyrus683) [inline solutions/ac-MultipleOf2019.mdx:74]: just `ans`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
these are just suggestions, if you don't want to do em that's ok

**Comment by SansPapyrus683** [comment]
also maybe rename it to `Main`? that way it'll work when you just chuck it in the OJ

File: solutions/ac-MultipleOf2019.mdx (hunk 1 above)
Comment (nathangong) [reply solutions/ac-MultipleOf2019.mdx:60]: num is used as the name of another variable

File: solutions/ac-MultipleOf2019.mdx (hunk 2 above)
Comment (brad182) [reply solutions/ac-MultipleOf2019.mdx:69]: come on this is silver im sure all people can understand that subtracting '0' gets you the actual number

**Comment by brad182** [comment]
> also maybe rename it to `Main`? that way it'll work when you just chuck it in the OJ

i don't know what OJ is

File: solutions/ac-MultipleOf2019.mdx (hunk 3 above)
Comment (brad182) [reply solutions/ac-MultipleOf2019.mdx:74]: come on that should be an optional fix

File: solutions/ac-MultipleOf2019.mdx (hunk 1 above)
Comment (brad182) [reply solutions/ac-MultipleOf2019.mdx:60]: numberString is more clear

File: solutions/ac-MultipleOf2019.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/ac-MultipleOf2019.mdx:69]: no?

**Comment by SansPapyrus683** [comment]
the ac online judge...

File: solutions/ac-MultipleOf2019.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/ac-MultipleOf2019.mdx:74]: why are you so pissed lmao

File: solutions/ac-MultipleOf2019.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/ac-MultipleOf2019.mdx:60]: yeah sure

File: solutions/ac-MultipleOf2019.mdx (hunk 2 above)
Comment (nathangong) [reply solutions/ac-MultipleOf2019.mdx:69]: Yeah I think the getNumericalValue method might be more understandable

File: solutions/ac-MultipleOf2019.mdx [hunk 4]
```diff
@@ -47,5 +47,41 @@ int main()
...
+		Kattio io = new Kattio();
+		String numberString = io.next();
+		int num = 0;
+		int stringLength = numberString.length();
+		int pow = 1;
+		int[] count = new int[2019];
+		count[0] = 1;
+
+		for (int i = stringLength - 1; i >= 0; i--) {
+			//find the remainder of the current number mod 2019
```
Comment (nwatx) [inline solutions/ac-MultipleOf2019.mdx:68]:
```suggestion
			// find the remainder of the current number mod 2019
```

File: solutions/ac-MultipleOf2019.mdx (hunk 2 above)
Comment (brad182) [reply solutions/ac-MultipleOf2019.mdx:69]: i feel like all of the courses like at alphastar include this idea of subracting characters in their intro to a programming language class

**Review by dongliu0426 — APPROVED** [review]

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/ac-MultipleOf2019.mdx [hunk 5]
```diff
@@ -47,5 +47,41 @@ int main()
...
+		int num = 0;
+		int stringLength = numberString.length();
+		int pow = 1;
+		int[] count = new int[2019];
+		count[0] = 1;
+
+		for (int i = stringLength - 1; i >= 0; i--) {
+			// find the remainder of the current number mod 2019
+			num = (num + pow * (numberString.charAt(i) - '0')) % 2019;
+			//increment the count of this remainder
```
Comment (nathangong) [inline solutions/ac-MultipleOf2019.mdx:70]: Also add spaces on the other comments for consistency
```suggestion
			// increment the count of this remainder
```

File: solutions/ac-MultipleOf2019.mdx [hunk 6]
```diff
@@ -47,5 +47,41 @@ int main()
...
+			count[num]++;
+			pow = pow * 10 % 2019;
+		}
+		long answer = 0;
+		for (int i = 0; i < 2019; i++) {
+			//there are nC2 ways to pair the numbers with the same remainder
```
Comment (nathangong) [inline solutions/ac-MultipleOf2019.mdx:76]:
```suggestion
			// there are nC2 ways to pair the numbers with the same remainder
```

**Review by nathangong — CHANGES_REQUESTED** [review]

**Review by nathangong — APPROVED** [review]

File: solutions/ac-MultipleOf2019.mdx [hunk 7]
```diff
@@ -47,5 +47,41 @@ int main()
...
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class MultipleOf2019 {
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
```
Comment (maggie-j-liu) [inline solutions/ac-MultipleOf2019.mdx:59]: add kattio codesnip

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**